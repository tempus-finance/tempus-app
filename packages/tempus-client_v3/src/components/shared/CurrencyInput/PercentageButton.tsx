import { FC, useCallback } from 'react';
import { NumberUtils } from 'tempus-core-services';
import ActionButton from '../ActionButton';

interface PercentageButtonProps {
  percentage: number;
  disabled?: boolean;
  onClick?: (percentage: number) => void;
}

const PercentageButton: FC<PercentageButtonProps> = props => {
  const { percentage, disabled, onClick } = props;
  const onButtonClick = useCallback(() => onClick?.(percentage), [percentage, onClick]);

  return (
    <ActionButton
      labels={{ default: NumberUtils.formatPercentage(percentage / 100), loading: '', success: '' }}
      onClick={onButtonClick}
      variant="secondary"
      size="small"
      state={disabled ? 'disabled' : 'default'}
    />
  );
};

export default PercentageButton;
