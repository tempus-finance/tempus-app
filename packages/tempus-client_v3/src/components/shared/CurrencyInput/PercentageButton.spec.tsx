import { fireEvent, render } from '@testing-library/react';
import PercentageButton, { PercentageButtonProps } from './PercentageButton';

const mockOnClick = jest.fn<void, [number]>();

const subject = (props: PercentageButtonProps) => render(<PercentageButton {...props} />);

describe('PercentageButton', () => {
  it('renders a percentage button', () => {
    const { getByRole, queryByText } = subject({ percentage: 50 });

    const actualButton = getByRole('button');
    const actualText = queryByText('50%');

    expect(actualButton).not.toBeNull();
    expect(actualText).not.toBeNull();

    expect(actualButton).toMatchSnapshot();
    expect(actualText).toMatchSnapshot();
  });

  it('renders a disabled percentage button', () => {
    const { getByRole } = subject({ percentage: 50, disabled: true });

    const actualButton = getByRole('button');

    expect(actualButton).not.toBeNull();
    expect(actualButton).toBeDisabled();

    expect(actualButton).toMatchSnapshot();
  });

  it('calls `onClick` when the percentage button is clicked', () => {
    const { getByRole } = subject({ percentage: 50, onClick: mockOnClick });

    const actualButton = getByRole('button');
    fireEvent.click(actualButton);

    expect(mockOnClick).toBeCalledTimes(1);
    expect(mockOnClick).toBeCalledWith(50);
  });
});
