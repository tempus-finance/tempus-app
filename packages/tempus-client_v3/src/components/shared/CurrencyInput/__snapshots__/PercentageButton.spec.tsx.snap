// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PercentageButton renders a disabled percentage button 1`] = `
<button
  class="tc__btn tc__actionButton tc__actionButton-border-secondary-small-disabled tc__actionButton-background-secondary-small-disabled tc__actionButton-hover-secondary-small-disabled tc__actionButton-shadow-disabled tc__actionButton-cursor-disabled tc__actionButton-size-small "
  data-selected="false"
  disabled=""
>
  <div
    class="tc__actionButton-status-label"
  >
    <div
      style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(122, 122, 122); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
    >
      50%
    </div>
  </div>
</button>
`;

exports[`PercentageButton renders a percentage button 1`] = `
<button
  class="tc__btn tc__actionButton tc__actionButton-border-secondary-small tc__actionButton-background-secondary-small tc__actionButton-hover-secondary-small tc__actionButton-shadow-default tc__actionButton-cursor-default tc__actionButton-size-small "
  data-selected="false"
>
  <div
    class="tc__actionButton-status-label"
  >
    <div
      style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
    >
      50%
    </div>
  </div>
</button>
`;

exports[`PercentageButton renders a percentage button 2`] = `
<div
  style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
>
  50%
</div>
`;
